lib/sensors/temperature.cpp

```diff
@@ -0,0 +1,39 @@
+#include "temperature.hpp"
+
+namespace hyped::sensors {
+
+Temperature::Temperature(hyped::io::I2c &i2c, hyped::core::ILogger &log) : log_(log), i2c_(i2c)
+{
+}
+
+std::optional<std::uint16_t> Temperature::read()
+{
+  if (i2c_.readByte(TEMPERATURE_1, STATUS) == 0) {
+    auto Temp_High = i2c_.readByte(TEMPERATURE_1, TEMP_H);
+    auto Temp_Low  = i2c_.readByte(TEMPERATURE_1, TEMP_L);
+    // getting the high and low values from the register
+
+    auto Temp = (TEMP_H << 8) | TEMP_L;
+    Temp      = Temp * 0.01;
+
+    return std::optional<std::uint16_t>{Temp};
+    // Reading from the Status register to see if the tempurature is ready to be read
+    // Need to have it do the correct effect and output a log of what occurred
+  } else {
+    return std::nullopt;
+  }
+  // log_.log(hyped::core::LogLevel::kFatal, "Temperature read not implemented");
+}
+
+bool Temperature::configure()
+{
+  i2c_.writeByte(TEMPERATURE_1, CTRL, 1);
```

Comment:
We want to know if this was successful or not, writeByte returns `hyped::core::Result`